SearchMicro/Sources/SearchMicro/Search/SearchViewModel.swift

```diff
@@ -28,24 +28,26 @@ public class SearchViewModel: ObservableObject {
     init() {
         $searchText
             .debounce(for: .seconds(0.5), scheduler: DispatchQueue.main)
-            .receive(on: DispatchQueue.main)
+            .receive(on: RunLoop.main)
             .sink { [weak self] value in
                 guard let self = self else { return }
                 self.debouncedText = value
                 
                 if (!self.debouncedText.isEmpty) {
-                    Task {
-                        do {
-                            let moviesResponse = try await self.movieService.getMovies(bySearchTerm: self.debouncedText) ?? []                            
-                            if moviesResponse.isEmpty {
+                    DispatchQueue.main.async {
```

Comment:
await MainActor.run { } --> Is this the same thing with  DispatchQueue.main.async {} , I am trying to understand how to use @MainActor. 

 RunLoop.main and  DispatchQueue.main --> also the same thing ? correct or how it different ?

Reply:
Should we also delete @MainActor on top on this ViewModel class.

Reply:
await MainActor.run is basically the same as DispatchQueue.main.async running on the Main loop asynchronously but does not work in the .sink response.  RunLoop.main and DispatchQueue.main are the same as well.  We can use either since both access the main thread asynchronously.  I was testing to see if the warning goes away using Runloop and it work not if I don't wrap the Task with DispatchQueue.main.async .